### DISCUSSION TITLE
504: GATEWAY_TIMEOUT

### DISCUSSION THREAD

**Discussion by Ashik045** [vercel/next.js#53664] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
This Serverless Function has timed out.

Your connection is working correctly.

Vercel is working correctly.
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Data fetching (gS(S)P, getInitialProps)

### Link to the code that reproduces this issue or a replay of the bug

https://vercel.com/ashik045/sociatek-client/logs?page=1&timeline=past30Minutes&startDate=1691383589035&endDate=1691385389035&selectedLogId=1691385366673791307102289550&selectedLogTimestamp=1691385366673&forceEndDate=1691385366673

### To Reproduce

I'm facing the server timeout issue on page that uses getServerSideProps on my application, it works but occasionally it feels like server went into sleep mode and resulting in 504 error, after several refreshes it wakes up and works as expected.

### Describe the Bug

I'm facing the server timeout issue on page that uses getServerSideProps on my application, it works but occasionally it feels like server went into sleep mode and resulting in 504 error, after several refreshes it wakes up and works as expected.

### Expected Behavior

It should run successfully because my server is working fine.

### Which browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

next start

**Comment by balazsorban44** [comment]
Hi, this sounds like you are having some slow code in that path or something similar. This issue tracker is for Next.js bugs, you could try opening a ticket with Vercel Support, or ask for tips in our [Discord](https://discord.com/invite/bUG2bvbtHy).
